### PR TITLE
Add C-grid capability for CICE6

### LINKED ISSUE #1271: Alllow CICE6 dynamics on the C-grid
## Description

Currently CICE6 dynamics are on the B-grid (cell corners). This is inconsistent with MOM6, which carries ocean velocities on the C-grid (N and E edges). This creates issues at isolated ocean points (along coastlines) or channels which are only 1-grid point wide. In these cases, there are no non-zero values with which to advect the ice on the B-grid. 

## Solution

Allow CICE6 dynamics to occur on the C-grid. When CICE6 dynamics are on the C-grid, the incompatibility between velocities located on the MOM6 C-grid and velocities located on the CICE6 B-grid is resolved.

### PR DESCRIPTION
**Pull request by DeniseWorthen**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [X] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [X] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [X] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsibility to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Updates CICE6 to include option to run dynamics on the C-grid. Additional variables are added to ``default_vars`` and ``ice_in_template`` to specify the grid staggering used. The following will specify the current configuration by default:

```
export GRIDATM=A
export GRIDOCN=A
export GRIDICE=B
```

To specify C-grid dynamics for CICE6, ``export GRIDICE=C`` should be used. A new test ``cpld_control_ciceC_p8`` is added.

### Issue(s) addressed

- fixes #1271 

## Testing

All baselines using CICE will change. This change is due solely to a change to how the 4 state variables (surface ocean u,v and surface tilts) are averaged. They will now be averaged as state variables instead of fluxes. See PR https://github.com/NOAA-EMC/CICE/pull/50 description for more details.

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [ ] jet.intel  (not available)
- [x] wcoss2.intel
- [NA] wcoss_cray
- [NA] wcoss_dell_p3
- [x] opnReqTest for newly added/changed feature: oRT log for new ciceC test was added at 1c3d42f
- [ ] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

- waiting on https://github.com/NOAA-EMC/CICE/pull/50

### CODE REVIEW

File: tests/default_vars.sh [hunk 1]
```diff
@@ -807,6 +807,10 @@ export RESTART_EXT=.false.
 export FRAZIL_FWSALT=.true.
 # default to write CICE average history files
 export CICE_HIST_AVG=.true.
+# default CICE B-grid, ATM and OCN are provided by cap on A grid
+export GRIDATM=A
+export GRIDOCN=A
+export GRIDICE=B
```
Comment (junwang-noaa) [inline tests/default_vars.sh:800]: @DeniseWorthen Are we going to have a feature test with GRIDICE=C?

Comment (DeniseWorthen) [reply tests/default_vars.sh:800]: Yes, I think we should. I've been trying to decide what to name it. I thought of ``cpld_ciceC_p8`` or ``cpld_control_ciceC_p8``. 

I've run the oRTs for both cpld_control and bmark_p8 for the C-grid, but I had to do it by just setting GRIDICE=C in default vars.

**Comment by DeniseWorthen** [comment]
@jkbk2004 This should be ready for RTs.

**Comment by jkbk2004** [comment]
> @jkbk2004 This should be ready for RTs.

sure! I will start from hera and cheyenne.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: BL
[BL] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/966000899/20220719184516/ufs-weather-model
[BL] Error: Test compile_009 failed in run_compile failed 
[BL] Error: Test compile_010 failed in run_compile failed 
[BL] Error: Test compile_011 failed in run_compile failed 
Please make changes and add the following label back: hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: gnu
Job: BL
[BL] Repo location: /glade/scratch/epicufsrt/autort/tests/auto/pr/966000899/20220719144514/ufs-weather-model
[BL] Error: Test compile_009 failed in run_compile failed 
[BL] Error: Test compile_010 failed in run_compile failed 
[BL] Error: Test compile_011 failed in run_compile failed 
Please make changes and add the following label back: cheyenne-gnu-BL

**Comment by jkbk2004** [comment]
@DeniseWorthen gnu picks up type mismatch issue at ice_dyn_shared.F90: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_720/compile_009 on hera

**Comment by DeniseWorthen** [comment]
I know. But I thought that was fixed before I had this merged into Consortium/main. I'm checking.

**Comment by DeniseWorthen** [comment]
@jkbk2004 I know what is wrong---I didn't update CICE after my fixes were merged at Consortium/main. I need to update the CICE hash. We will need to start over. Is that OK, or do you want to move on?

**Comment by jkbk2004** [comment]
> @jkbk2004 I know what is wrong---I didn't update CICE after my fixes were merged at Consortium/main. I need to update the CICE hash. We will need to start over. Is that OK, or do you want to move on?

Sure! go ahead to update. jet is down. so test will continue tomorrow, I guess.

**Comment by DeniseWorthen** [comment]
Can we kill whatever is currently running on cheyenne.intel and hera.intel? The way we'd kill everything on cheyenne is 

``qselect -u USERNAME | xargs qdel``

**Comment by jkbk2004** [comment]
> Can we kill whatever is currently running on cheyenne.intel and hera.intel? The way we'd kill everything on cheyenne is
> 
> `qselect -u USERNAME | xargs qdel`

let me try to cancel jobs on hera and cheyenne. I will keep you posted.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
[BL] Repo location: /glade/scratch/epicufsrt/autort/tests/auto/pr/966000899/20220719150020/ufs-weather-model
[BL] Error: Test cpld_control_c192_p8 002 failed in run_test failed 
[BL] Error: Test cpld_bmark_p8 003 failed in run_test failed 
[BL] Error: Test hafs_regional_atm 076 failed in run_test failed 
[BL] Error: Test hafs_regional_atm_thompson_gfdlsf 077 failed in run_test failed 
[BL] Error: Test hafs_regional_atm_wav 079 failed in run_test failed 
[BL] Error: Test hafs_regional_atm_ocn_wav 080 failed in run_test failed 
[BL] Error: Test hafs_regional_1nest_atm 081 failed in run_test failed 
[BL] Error: Test hafs_regional_telescopic_2nests_atm 082 failed in run_test failed 
[BL] Error: Test hafs_global_1nest_atm 083 failed in run_test failed 
[BL] Error: Test hafs_global_multiple_4nests_atm 084 failed in run_test failed 
[BL] Error: Test hafs_regional_specified_moving_1nest_atm 085 failed in run_test failed 
[BL] Error: Test hafs_regional_storm_following_1nest_atm 086 failed in run_test failed 
[BL] Error: Test hafs_regional_storm_following_1nest_atm_ocn 087 failed in run_test failed 
[BL] Error: Test hafs_regional_storm_following_1nest_atm_ocn_wav 088 failed in run_test failed 
[BL] Error: Test hafs_global_storm_following_1nest_atm 089 failed in run_test failed 
[BL] Error: Test hafs_regional_docn 090 failed in run_test failed 
[BL] Error: Test hafs_regional_docn_oisst 091 failed in run_test failed 
[BL] Error: Test hafs_regional_datm_cdeps 092 failed in run_test failed 
[BL] Error: Test atmaero_control_p8 105 failed in run_test failed 
[BL] Error: Test regional_atmaq 107 failed in run_test failed 
Please make changes and add the following label back: cheyenne-intel-BL

**Comment by jkbk2004** [comment]
@DeniseWorthen I think I canceled all jobs on hera and cheyenne. I will double check and start tests with gnu first again.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: BL
[BL] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/966000899/20220719190020/ufs-weather-model
[BL] Error: Test hafs_regional_specified_moving_1nest_atm 085 failed in run_test failed 
[BL] Error: Test hafs_regional_storm_following_1nest_atm_ocn 087 failed in run_test failed 
[BL] Error: Test hafs_regional_storm_following_1nest_atm_ocn_wav 088 failed in run_test failed 
[BL] Error: Test hafs_global_storm_following_1nest_atm 089 failed in run_test failed 
[BL] Error: Test hafs_regional_docn 090 failed in run_test failed 
[BL] Error: Test hafs_regional_docn_oisst 091 failed in run_test failed 
[BL] Error: Test datm_cdeps_mx025_cfsr 099 failed in run_test failed 
[BL] Error: Test datm_cdeps_mx025_gefs 100 failed in run_test failed 
[BL] Error: Test atmaero_control_p8 105 failed in run_test failed 
[BL] Error: Test regional_atmaq 107 failed in run_test failed 
Please make changes and add the following label back: hera-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: gnu
Job: BL
[BL] Repo location: /glade/scratch/epicufsrt/autort/tests/auto/pr/966000899/20220719183014/ufs-weather-model
[BL] Error: Test cpld_control_noaero_p8 032 failed in run_test failed 
[BL] Error: Test cpld_control_nowave_noaero_p8 033 failed in run_test failed 
[BL] Error: Test cpld_debug_noaero_p8 034 failed in run_test failed 
[BL] Error: Test datm_cdeps_control_cfsr 035 failed in run_test failed 
Please make changes and add the following label back: cheyenne-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: BL
[BL] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/966000899/20220720003012/ufs-weather-model
[BL] Error: Test cpld_control_noaero_p8 032 failed in run_test failed 
[BL] Error: Test cpld_control_nowave_noaero_p8 033 failed in run_test failed 
[BL] Error: Test cpld_debug_noaero_p8 034 failed in run_test failed 
[BL] Error: Test datm_cdeps_control_cfsr 035 failed in run_test failed 
Please make changes and add the following label back: hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
[BL] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/966000899/20220719193012/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/966000899/20220719205339/ufs-weather-model
[RT] Error: Test cpld_control_ciceC_p8 006 failed in check_result failed 
[RT] Error: Test cpld_control_ciceC_p8 006 failed in run_test failed 
Please make changes and add the following label back: orion-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
[BL] Repo location: /glade/scratch/epicufsrt/autort/tests/auto/pr/966000899/20220719184514/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /glade/scratch/epicufsrt/autort/tests/auto/pr/966000899/20220719201349/ufs-weather-model
[RT] Error: Test cpld_control_ciceC_p8 006 failed in run_test failed 
Please make changes and add the following label back: cheyenne-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: BL
[BL] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/966000899/20220720004510/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/966000899/20220720021445/ufs-weather-model
[RT] Error: Test cpld_control_ciceC_p8 006 failed in check_result failed 
[RT] Error: Test cpld_control_ciceC_p8 006 failed in run_test failed 
Please make changes and add the following label back: hera-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: BL
[BL] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/966000899/20220720003009/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/966000899/20220720044107/ufs-weather-model
[RT] Error: Test cpld_control_ciceC_p8 006 failed in check_result failed 
[RT] Error: Test cpld_control_ciceC_p8 006 failed in run_test failed 
Please make changes and add the following label back: gaea-intel-BL

**Comment by jkbk2004** [comment]
@DeniseWorthen sounds like people running jobs on jet. I will run manually on jet now.

**Comment by jkbk2004** [comment]
> @DeniseWorthen sounds like people running jobs on jet. I will run manually on jet now.

@DeniseWorthen network on jet is not going thru. sounds only existing jobs are running.

**Comment by DeniseWorthen** [comment]
@jkbk2004 Do you want to wait until Jet is available? I think it is OK to skip a platform for a known reason like this. The next PR is MOM6 and it will also change baselines; only if someone tried to run this specific commit on jet would they have an issue.

**Comment by jkbk2004** [comment]
@DeniseWorthen @BrianCurtis-NOAA I agree to skip jet. I like to move on to merge in the pr.

**Review by jiandewang — APPROVED** [review]

**Comment by jkbk2004** [comment]
I am in the workshop at college park. Going out to have lunch. I will be back in an hour.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by jkbk2004 — APPROVED** [review]

**Comment by jkbk2004** [comment]
can we start merging?

**Comment by DeniseWorthen** [comment]
CICE has been updated and the gitmodule reverted. This is ready for commit.

**Pull request merged**